src/main/kotlin/io/zerobase/smarttracing/GraphDao.kt

```diff
@@ -53,4 +57,183 @@ class GraphDao(private val driver: Driver) {
+        driver.session().use {
+            it.writeTransaction { txn ->
+                txn.run(
+                        """
+                        MATCH (o:Organization { id: '${id}' })
+                        SET o.multisite = 'true'
+                        """
+                )
+            }
+        }
+    }
+
+    /**
+     * Creates site.
+     *
+     * @param id organization id.
+     * @param name name of the site.
+     * @param category category of the site.
+     * @param subcategory subcategory of the site.
+     * @param lat latitude of the site.
+     * @param long longitude of the site.
+     * @param testing whether the site can preform testing.
+     * @param phone contact phone of site manager
+     * @param email contact email of site manager
+     * @param contactName contact name of site manager
+     */
+    fun createSite(id: String, name: String, category: String,
+                   subcategory: String, lat: Float, long: Float,
+                   testing: Boolean, phone: String?, email: String?,
+                   contactName: String?): SiteId? {
+        val query = if (phone == null) {
+            """
+            MATCH (o:Organization { id: '${id}' })
+            CREATE (s:Site {
+                        id: '${UUID.randomUUID()}',
+                        phone: o.phone,
+                        email: o.email,
+                        contactName: o.contactName,
+                        name: '${name}',
+                        category: '${category}',
+                        subcategory: '${subcategory}',
+                        latitude: '${lat}',
+                        longitude: '${long}',
+                        testing: '${testing}'
+            })
+            CREATE (o)-[r:OWNS]->(s)
+            RETURN s.id as id
+            """.trimIndent()
+        } else {
+             """
+            MATCH (o:Organization { id: '${id}' })
+            CREATE (s:Site {
+                        id: '${UUID.randomUUID()}',
+                        phone: '${phone!!}',
+                        email: '${email!!}',
+                        contactName: '${contactName!!}',
+                        name: '${name}',
+                        category: '${category}',
+                        subcategory: '${subcategory}',
+                        latitude: '${lat}',
+                        longitude: '${long}',
+                        testing: '${testing}'
+            })
+            CREATE (o)-[r:OWNS]->(s)
+            RETURN s.id as id
+            """.trimIndent()
+        }
+        return driver.session().use {
+            it.writeTransaction { txn ->
+                val result = txn.run(query).single()["id"].asString()
+                return@writeTransaction result?.let { SiteId(it) }
+            }
+        }
+    }
+
+    /**
+     * Gets all the sites list
+     *
+     * @param id id of the organization
+     */
+    fun getSites(id: String): List<SiteResponse>? {
```

Comment:
in general, i'd rather separate the API DTO models from the database. having a DAO return an API DTO is too much coupling. define a model for a site and let the resource convert it into a DTO.

Reply:
sure i will move that over there